watch_service.go

```diff
@@ -67,9 +72,12 @@ func (ws *watchService) watchFile(ctx context.Context, projectName, repoName, la
 func (ws *watchService) watchRepo(ctx context.Context,
 	projectName, repoName, lastKnownRevision, pathPattern string, timeout time.Duration) <-chan *WatchResult {
 	watchResult := make(chan *WatchResult)
-	if len(pathPattern) != 0 && !strings.HasPrefix(pathPattern, "/") {
-		// Normalize the pathPattern when it does not start with "/" so that the pathPattern fits into the url.
-		pathPattern = "/**/" + pathPattern
+
+	// Normalize pathPattern
+	if len(pathPattern) == 0 {
+		pathPattern = "/**"
+	} else if !strings.HasPrefix(pathPattern, "/") {
+		pathPattern = "/" + pathPattern
```

Comment:
`pathPattern = "/**/" + pathPattern`

Reply:
If a user specified a path pattern that did not start with `/`, path pattern can be normalized into `/**/<pathPattern>`. The proposed change does not prepend `/**/` to `/foo/*.json` because it starts with `/`.

Reply:
For example, the proposed change is to transform the patterns like `*.json` to `/**/*.json` which is practically same while making sure the path pattern always starts with `/`.

Reply:
If we did not prepend `/**/`, the URL will be like: `.../contents*.json` rather than `.../contents/**/*.json`.

Reply:
Yeah we should add `/**/` when the `pathPattern` is not empty and doesn't start with `/`. :)

Reply:
> f we did not prepend `/**/`, the URL will be like: `.../contents*.json` rather than `.../contents/**/*.json`.

I think in this case, the output will be `/*.json`. So please correct me if i am wrong: with path `/*.json`, won't recursively find all json files?

If yes, I agree that we should add `/**/` instead

Reply:
`/*.json` will find the `*.json` at the root directory only.

Reply:
@trustin san, so the api document:
```
A path pattern is a variant of glob: 
"/**" - find all files recursively
"*.json" - find all JSON files recursively
```
Is wrong?

Reply:
No, it's correct. What we are dealing with here is how our REST API is designed. For example, when using Thrift API, you don't have to handle this at all. It's just because we have to make sure all path patterns start with `/` in REST API because otherwise the request path is broken / ugly.

Reply:
@trustin san, I have update as your comment. PTAL 🙇